testdata/bench_literals/main.go

```diff
@@ -0,0 +1,12 @@
+package main
+
+import "testing"
+
+func main() {
+	res := testing.Benchmark(func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			run() //  generated automatically in bench_literals_test.go
+		}
+	})
+	println(res.NsPerOp(), res.AllocsPerOp(), res.AllocedBytesPerOp())
```

Comment:
interesting way of using testing.Benchmark - but note that I think this is slightly broken, as you get a new testing.B, so parameters like b.N will be reset to the default values. Flags like `-benchtime` will likely not work properly.

I'm not sure what the best approach would be. Something that comes to mind is to template a main program that uses lots of literals, and each iteration of the benchmark would simply run the program. This is the simplest option, and it would require just a few changes to your code, but the benchmark will remain a black box (e.g. not able to grab cpu or memory profiles), and the benchmark will always count the exec overhead.

An alternative is to run the benchmark via a short Go program rather than `go test -bench`. The program would:

* Template a `*_test.go` with a benchmark, and the same "run" func you fill above
* Use `garble test -bench` to obfuscate the benchmark code and run the benchmark

This approach doesn't have the disadvantages I mentioned before. It will require being run with a special `go run bench.go` or somesuch, but I think that's OK.